lib/loader/file_loader.rb

```diff
@@ -36,7 +36,8 @@ def skip_matching_header(filehandle, skip_match)
     # @return [Integer] Number of records loaded
     def load(filename, filehandle: Zinzout.zin(filename), skip_header_match: nil)
       logger.info "Loading #{filename}, batches of #{ppnum @batch_size}"
-      skip_matching_header(filehandle, skip_header_match).each.lazy
+
+      skip_matching_header(filehandle.enum_for(:each), skip_header_match).lazy
```

Comment:
this change ensures we get an enumerator back regardless of what filehandle was to begin with -- all we need is for it to respond to `:each`